playbooks/common/openshift-master/config.yml

```diff
@@ -122,6 +122,7 @@
       openshift_master_certs_no_etcd:
       - admin.crt
       - master.kubelet-client.crt
+      - "{{ 'master.proxy-client.crt' if (openshift.common.version != '' and (openshift.common.version | version_compare('1.0.6', '>') and openshift.common.version | version_compare('3.0','<') ) or openshift.common.version | version_compare('3.0.2','>')) else omit }}"
```

Comment:
This conditional is a bit hard to follow...  might want to use a combination of deployment_type and version instead of testing > 1.0.6 and < 3.0

either way, it is definitely going to be hairy...  might want to use set_fact to set a variable to use to conditionally omit the master.proxy-client.crt.